public/app/plugins/datasource/prometheus/add_label_to_query.ts

```diff
@@ -67,11 +56,19 @@ export function addLabelToQuery(
 
   while (match) {
     const prefix = query.slice(lastIndex, match.index);
-    const selector = match[1];
-    const selectorWithLabel = addLabelToSelector(selector, key, transformedValue, operator);
-    lastIndex = match.index + match[1].length + 2;
+    lastIndex = match.index + match[2].length + 2;
     suffix = query.slice(match.index + match[0].length);
-    parts.push(prefix, selectorWithLabel);
+    // If we matched 1st group, we know it is Grafana's variable and we don't want to add labels
+    if (match[1]) {
```

Comment:
This is unfortunately necessary because of negative lookbehind not being supported in safari.